src/integrations/front-end/print-qrcode-render.php

```diff
@@ -43,9 +43,11 @@ public function generate_qr_code() {
 			return;
 		}
 
-		$nonce = \filter_input( INPUT_GET, 'nonce', FILTER_SANITIZE_STRING );
-		if ( ! \wp_verify_nonce( $nonce, 'yoast_seo_qr_code' ) ) {
-			\wp_die( 'This is not a QR code endpoint for public consumption.' );
+		$code = \filter_input( INPUT_GET, 'code', FILTER_SANITIZE_STRING );
+		if ( \wp_hash( $url ) !== $code ) {
```

Comment:
I think hash_equals() should be preferred than `!==` the strings, to prevent timing attacks